std/traits.d

```diff
@@ -1495,6 +1499,193 @@ private FunctionAttribute extractAttribFlags(Attribs...)()
     return res;
 }
 
+/**
+Checks whether a function has the given attributes attached.
+
+Params:
+    func = function to check
+    attributes = variadic number of function attributes as strings
+
+Returns:
+    `true`, if the function has the list of attributes attached and `false` otherwise.
+
+See_Also:
+    $(LREF functionAttributes)
+*/
+template hasFunctionAttributes(args...)
+    if (isCallable!(args[0]) && args.length > 0
+         && allSatisfy!(isSomeString, typeof(args[1 .. $])))
+{
+    enum bool hasFunctionAttributes = {
+        import std.algorithm.searching : canFind;
+        import std.range : only;
+        enum funcAttribs = only(__traits(getFunctionAttributes, args[0]));
+        auto b = true;
+        foreach (attribute; args[1..$])
+        {
+            if (!funcAttribs.canFind(attribute))
+            {
+                b = false;
+                break;
+            }
+        }
+        return b;
+    }();
+}
+
+///
+unittest
+{
+    real func(real x) pure nothrow @safe
+    {
+        return x;
```

Comment:
In order to avoid false coverage reports, you may want to just declare the function without implementing it.